### PR TITLE
Refactor call-to-action  component test file to new format

### PR DESCRIPTION
**Pull request by precious-onyenaucheya-ons**
### What is the context of this PR?

Fixes [#140](https://github.com/ONSdigital/design-team/issues/140) on the design system team board

### How to review this PR

- Do we need any additional tests?
- Are the tests descriptions clear?
- Does the order of the test document make sense?
- Is it internally consistent?
- Is it consistent with the [guidelines for test refactoring](https://confluence.ons.gov.uk/display/SDC/Testing+Refactor+Guidance)?
- run this test command to view the test results:
- yarn test --testNamePattern="FOR: call-to-action"

### Checklist

This needs to be completed by the person raising the PR.

-   [x] I have selected the correct Assignee
-   [x] I have linked the correct Issue

### CODE REVIEW

File: src/components/call-to-action/_macro.spec.js [hunk 1]
```diff
@@ -3,46 +3,35 @@
...
+    describe('GIVEN: Params: default', () => {
+        describe('WHEN: params are at default', () => {
+            const $ = cheerio.load(renderComponent('call-to-action', EXAMPLE_CALL_TO_ACTION));
+
+            test('THEN: it passes jest-axe checks', async () => {
+                const results = await axe($.html());
+                expect(results).toHaveNoViolations();
+            });
+
+            test('THEN: it has the provided `headingText`', () => {
```
Comment (rmccar) [inline src/components/call-to-action/_macro.spec.js:19]:
```suggestion
            test('THEN: it has the provided headingText', () => {
```

File: src/components/call-to-action/_macro.spec.js [hunk 2]
```diff
@@ -3,46 +3,35 @@
...
+                const headingText = $('.ons-call-to-action__heading').text().trim();
+                expect(headingText).toBe('Call to action heading.');
+            });
+
+            test('THEN: it has the provided `paragraphText`', () => {
```
Comment (rmccar) [inline src/components/call-to-action/_macro.spec.js:24]:
```suggestion
            test('THEN: it has the provided paragraphText', () => {
```

File: src/components/call-to-action/_macro.spec.js [hunk 3]
```diff
@@ -3,46 +3,35 @@
...
+                const paragraphText = $('.ons-call-to-action__text').text().trim();
+                expect(paragraphText).toBe('Descriptive text about call to action');
+            });
+
+            test('THEN: it outputs the expected call-to-action button', () => {
+                const buttonText = $('.ons-btn__text').text().trim();
+
+                expect(buttonText).toBe('Start');
+                expect($('.ons-btn--link').attr('href')).toBe('https://example.com/start');
```
Comment (rmccar) [inline src/components/call-to-action/_macro.spec.js:33]: May as well do here what you have done with the buttonText const above

File: src/components/call-to-action/_macro.spec.js [hunk 4]
```diff
@@ -3,46 +3,36 @@
...
-
-        expect(buttonSpy.occurrences[0]).toHaveProperty('text', 'Start');
-        expect(buttonSpy.occurrences[0]).toHaveProperty('url', 'https://example.com/start');
+import { renderComponent } from '../../tests/helpers/rendering';
+import { EXAMPLE_CALL_TO_ACTION } from './_test-examples';
+
+describe('FOR: call-to-action', () => {
+    describe('GIVEN: Params: default', () => {
+        describe('WHEN: params are at default state', () => {
+            const $ = cheerio.load(renderComponent('call-to-action', EXAMPLE_CALL_TO_ACTION));
```
Comment (rmccar) [inline src/components/call-to-action/_macro.spec.js:12]: These aren't defaults, they are params being passed in

**Review by alessioventuriniAND — COMMENTED** [review]
I don't think we need to do this as the call to action component is being removed with our next major release. I think you can close this pr and close the ticket too

File: src/components/call-to-action/_test-examples.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+export const EXAMPLE_CALL_TO_ACTION = {
```
Comment (balibirchlee) [inline src/components/call-to-action/_test-examples.js:1]: this should be just the required params

File: src/components/call-to-action/_macro.spec.js [hunk 6]
```diff
@@ -3,46 +3,36 @@
...
-
-        faker.renderComponent('call-to-action', EXAMPLE_CALL_TO_ACTION);
-
-        expect(buttonSpy.occurrences[0]).toHaveProperty('text', 'Start');
-        expect(buttonSpy.occurrences[0]).toHaveProperty('url', 'https://example.com/start');
+import { renderComponent } from '../../tests/helpers/rendering';
+import { EXAMPLE_CALL_TO_ACTION } from './_test-examples';
+
+describe('FOR: call-to-action', () => {
+    describe('GIVEN: Params: default', () => {
```
Comment (balibirchlee) [inline src/components/call-to-action/_macro.spec.js:10]: this should be required or all depending on what you are testing against - and should only include the jest-axe test

File: src/components/call-to-action/_macro.spec.js [hunk 7]
```diff
@@ -3,46 +3,36 @@
...
+        describe('WHEN: params are at default state', () => {
+            const $ = cheerio.load(renderComponent('call-to-action', EXAMPLE_CALL_TO_ACTION));
+
+            test('THEN: it passes jest-axe checks', async () => {
+                const results = await axe($.html());
+                expect(results).toHaveNoViolations();
+            });
+
+            test('THEN: it has the provided headingText', () => {
```
Comment (balibirchlee) [inline src/components/call-to-action/_macro.spec.js:19]: these should each have a new given for the param that they are covering

File: src/components/call-to-action/_macro.spec.js [hunk 8]
```diff
@@ -3,46 +3,36 @@
...
+                const headingText = $('.ons-call-to-action__heading').text().trim();
+                expect(headingText).toBe('Call to action heading.');
+            });
+
+            test('THEN: it has the provided paragraphText', () => {
+                const paragraphText = $('.ons-call-to-action__text').text().trim();
+                expect(paragraphText).toBe('Descriptive text about call to action');
+            });
+
+            test('THEN: it outputs the expected call-to-action button', () => {
```
Comment (balibirchlee) [inline src/components/call-to-action/_macro.spec.js:29]: button should be a given - then you can either have one when for the text and url (as in this case they are both required) or one when for each

**Review by balibirchlee — CHANGES_REQUESTED** [review]

**Pull request closed without merging**